File: python/2_testcases_test.py
import pytest


def merge_2_lists(listA, listB):
    mergedList = []
    mergedList.extend(a for a in listA if a not in mergedList)
    mergedList.extend(b for b in listB if b not in mergedList)
    return mergedList


@pytest.mark.parametrize('listA,listB,expected_result', [
  (['A', 'B'], ['C', 'D'], ['A', 'B', 'C', 'D']),
  (['A', 'A'], ['A', 'B'], ['A', 'B']),
  ([], ['C'], ['C']),
  ([], [], [])
])
def test_with_cases(listA, listB, expected_result):
    result = merge_2_lists(listA, listB)
    assert result == expected_result

File: js/2_testcasesTest.js
const assert = require('assert');

function merge2Lists(listA, listB) {
    const withDuplicates = listA.concat(listB);
    const mergedList = Array.from(new Set(withDuplicates));
    return mergedList;
}

describe('Test that the lists are correctly merged', function () {
    const tests = [
        {listA: ['A', 'B'], listB: ['C', 'D'], expected: ['A', 'B', 'C', 'D']},
        {listA: ['A', 'A'], listB: ['A', 'B'], expected: ['A', 'B']},
        {listA: [], listB: ['C'], expected: ['C']},
        {listA: [], listB: [], expected: []}
    ];

    tests.forEach(function(test) {
        it('should return list containing ' + test.expected, function () {
            const result = merge2Lists(test.listA, test.listB);
            assert.deepEqual(result, test.expected);
        });
    });
});
